### PR TITLE
Separate test into versions as first step

### PR DESCRIPTION
**Pull request by sergicastro**
As version increases new tests are added so, we cannot run the same
tests for all the versions.
This way we can control in different files the tests that are needed for
each minor version.

### CODE REVIEW

**Comment by sergicastro** [comment]
also added more gke kubernetes versions to run the e2e.

File: .github/workflows/e2e_tests.yml [hunk 1]
```diff
@@ -83,35 +83,24 @@ jobs:
         run: echo ::set-output name=TAG::${GITHUB_REF/refs\/tags\/}
         shell: bash
 
+      - name: Get minor version
+        id: get_minor_ver
+        run: echo ::set-output name=MINORVER::$(echo $TAG | grep -Eo '[0-9]+\.[0-9]+')
```
Comment (deva26) [inline .github/workflows/e2e_tests.yml:88]: Can we name it REL_BRANCH_VER instead of MINORVER? Here and elsewhere.

File: tetrateci/test_1.7.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
+set -e
+
+if [[ ${CLUSTER} == "gke" ]]; then
+  # Overlay CNI Parameters for GCP : https://github.com/tetratelabs/getistio/issues/76
+  pip install pyyaml --user && ./tetrateci/gen_iop.py
+  CLUSTERFLAGS="-istio.test.kube.helm.iopFile $(pwd)/tetrateci/iop-gke-integration.yml"
+fi
+
+go test -tags=integ ./tests/integration/operator/...  -p 1  -test.v
```
Comment (deva26) [inline tetrateci/test_1.7.sh:9]: Can you cross check if the same set of tests work for all distros?

Comment (sergicastro) [reply tetrateci/test_1.7.sh:9]: yes, I wish they do and it's what I'm testing.

Comment (sergicastro) [reply tetrateci/test_1.7.sh:10]: Finally, I needed to add [this commit](https://github.com/tetratelabs/istio/pull/20/commits/fd0e42164f364d874086bd5a44cabe62614df6dc) to run some tests depending on the distros.

File: .github/workflows/e2e_tests.yml [hunk 3]
```diff
@@ -126,7 +115,13 @@ jobs:
 
     strategy:
       matrix:
-        version: ["1.17.14-gke.2600"]
+        include:
+          - version: "1.17.14-gke.2600"
+            channel: "regular"
+          - version: "1.16.15-gke.6000"
+            channel: "stable"
+          - version: "1.18.12-gke.1201"
```
Comment (deva26) [inline .github/workflows/e2e_tests.yml:123]: nit - can we re-arrange from higher version to lower for easy readability

Comment (sergicastro) [reply .github/workflows/e2e_tests.yml:123]: yes, will also add the versions for EKS (and AKS when test pass)

File: .github/workflows/e2e_tests.yml [hunk 4]
```diff
@@ -273,7 +248,7 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} 
         with:
           tag_name: ${{ steps.get_tag.outputs.TAG }}
-          release_name: Tetrate Istio 
+          release_name: Get Istio
```
Comment (deva26) [inline .github/workflows/e2e_tests.yml:251]: nit - Change it to GetIstio

File: .github/workflows/e2e_tests.yml [hunk 5]
```diff
@@ -126,7 +115,13 @@ jobs:
 
     strategy:
```
Comment (deva26) [inline .github/workflows/e2e_tests.yml:116]: For now add needs on eks-test and aks-test for gke. And add a note and a TODO why this is needed. And move the gke job to below AKS

**Review by deva26 — APPROVED** [review]
Thanks, LGTM. minor nits.

**Review by deva26 — CHANGES_REQUESTED** [review]
@sergicastro --- request from product to keep tetrate in the version name. There is some confusion around it. Need to hold on changing tetrate to getistio

**Comment by sergicastro** [comment]
So, I need to revert all the renaming commit or just change [this line](https://github.com/tetratelabs/istio/pull/20/files#diff-37aed4ad4a8c3a4220c5e3c2eee33093073b8d64f480cc742b63764c859b6c72R252)?

**Review by deva26 — APPROVED** [review]

**Pull request merged**